src/pages/d/[id].tsx

```diff
@@ -49,11 +50,8 @@ export default function Home() {
       });
 
       const defaultUserInfo = randUserInfo();
-      await r.mutate.initClientState({
-        id: await r.clientID,
-        defaultUserInfo,
-      });
-      await r.mutate.initShapes();
+      await r.mutate.initClientState(defaultUserInfo);
```

Comment:
Consider the case of a new client connecting to an existing room:

- This mutator runs optimistically on a fresh cache. It sees that there is no `initialized` flag set and creates a default set of shapes.
- The user sees five random shapes optimistically
- Then the mutator runs on server
- In this case. the initialized flag is already set, the default shapes aren't picked and the existing shapes remain
- The client syncs and the user sees the shapes suddenly change to a different set

For some reason when I use replidraw-do I don't see this effect. Perhaps something somewhere else is delaying rendering until first sync somehow, or perhaps this is some artifact of the early version of Reflect this app is using that I've forgotten about.

In any case, we should demonstrate a good pattern for initialization. An easy way is to make `initShapes` server-only, using your new env flag.

```ts
mutators: {
  initShapes: async (tx: WriteTransaction, shapes: Shape[]) => {
    if (tx.env != 'server') {
      return;
    }
    if ((await tx.get("initialized")) === true) {
      return;
    }
    ...
  }
}
```